minikubek8s.sh

```diff
@@ -0,0 +1,171 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tiller
+  namespace: kube-system
+EOF
+
+# Create Dir for yml files
+echo "Creating Folder base for yml files........................"
+mkdir base
+mv ./service-account.yml base/
+
+# Apply service account
+echo "Apply Service-account....................................."
+kubectl apply -f base/service-account.yml
+kubectl get serviceaccounts -n kube-system
+
+# Create yml role-binding config for service account tiller 
+cat <<EOF >./role-binding.yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: tiller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+  - kind: ServiceAccount
+    name: tiller
+    namespace: kube-system
+EOF
+
+# Apply role binding
+mv ./role-binding.yml base/ 
+kubectl apply -f base/role-binding.yml
+kubectl get clusterrolebindings.rbac.authorization.k8s.io
+
+# install helm
+echo "------------------------------helm checking-----------------------------"
+export helm=$(which helm)
+if [ -f "$helm" ]; then
+    echo "$helm already exist......................................."
+else
+    curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get > install-helm.sh
+    chmod u+x install-helm.sh
+    ./install-helm.sh
+    helm version
+fi
+
+# Deploy tiller
+echo "Deploy tiller............................................."
+helm init --service-account tiller --wait
+kubectl get pods -n kube-system
+
+# Create namespace monitoring
+cat <<EOF >./namespace.yml
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: monitoring
+EOF
+
+# Apply namespace
+mv ./namespace.yml base/
+kubectl apply -f base/namespace.yml
+kubectl get namespaces
+
+# install prometheus with helm
+echo "install prometheus with helm.............................."
+sudo helm repo update
+sudo helm install stable/prometheus --namespace monitoring --name prometheus
+kubectl get pods -n monitoring
+
+# Create a Prometheus data source config map
+cat <<EOF >./configmap.yml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-grafana-datasource
+  namespace: monitoring
+  labels:
+    grafana_datasource: '1'
+data:
+  datasource.yaml: |-
+    apiVersion: 1
+    datasources:
+    - name: Prometheus
+      type: prometheus
+      access: proxy
+      orgId: 1
+      url: http://prometheus-server.monitoring.svc.cluster.local
+EOF
+
+# Apply configmap
+mv ./configmap.yml base/
+kubectl apply -f base/configmap.yml
+kubectl get configmaps -n monitoring
+
+# Create values.yml for override grafana value
+cat <<EOF >./values.yml
+sidecar:
+  datasources:
+    enabled: true
+    label: grafana_datasource
+EOF
+
+mv ./values.yml base/
+
+# Install grafana with helm and override grafana value
+echo "install grafana with helm................................."
+sudo helm install stable/grafana -f base/values.yml --namespace monitoring --name grafana
+kubectl get pods -n monitoring
```

Comment:
If grana and prometheus take some time to spin up, I will suggest to include some:
echo "Waiting X seconds to allow grafana to start"
sleep X

Measure with your computer and your connection how long can it take.

Reply:
Also, take a look to previous comment. Try to write only the minimum number of lines! 
If `kubectl get pods -n monitoring` is not going to serve any purpose, just delete it.